dlt/destinations/impl/sink/configuration.py

```diff
@@ -0,0 +1,71 @@
+from typing import TYPE_CHECKING, Optional, Final, Callable, Union, Any
+from importlib import import_module
+
+from dlt.common.configuration import configspec
+from dlt.common.destination import TLoaderFileFormat
+from dlt.common.destination.reference import (
+    DestinationClientConfiguration,
+    CredentialsConfiguration,
+)
+from dlt.common.typing import TDataItems
+from dlt.common.schema import TTableSchema
+from dlt.common.configuration.exceptions import ConfigurationValueError
+
+
+TSinkCallable = Callable[[Union[TDataItems, str], TTableSchema], None]
+
+
+@configspec
+class SinkClientCredentials(CredentialsConfiguration):
+    callable: Optional[str] = None  # noqa: A003
+
+    def parse_native_representation(self, native_value: Any) -> None:
+        # a callable was passed in
+        if callable(native_value):
+            self.resolved_callable: TSinkCallable = native_value
+        # a path to a callable was passed in
+        if isinstance(native_value, str):
+            self.callable = native_value
+
+    def to_native_representation(self) -> Any:
+        return self.resolved_callable
+
+    def on_resolved(self) -> None:
+        if self.callable:
+            try:
+                module_path, attr_name = self.callable.rsplit(".", 1)
+                dest_module = import_module(module_path)
+            except ModuleNotFoundError as e:
+                raise ConfigurationValueError(
+                    f"Could not find callable module at {module_path}"
+                ) from e
+            try:
+                self.resolved_callable = getattr(dest_module, attr_name)
+            except AttributeError as e:
+                raise ConfigurationValueError(
+                    f"Could not find callable function at {self.callable}"
+                ) from e
+
+        if not hasattr(self, "resolved_callable"):
+            raise ConfigurationValueError("Please specify callable for sink destination.")
+
+        if not callable(self.resolved_callable):
+            raise ConfigurationValueError("Resolved Sink destination callable is not a callable.")
+
+
+@configspec
+class SinkClientConfiguration(DestinationClientConfiguration):
+    destination_type: Final[str] = "sink"  # type: ignore
+    credentials: SinkClientCredentials = None
+    loader_file_format: TLoaderFileFormat = "puae-jsonl"
+    batch_size: int = 10
+
+    if TYPE_CHECKING:
+
+        def __init__(
+            self,
+            *,
+            credentials: Union[SinkClientCredentials, TSinkCallable, str] = None,
+            loader_file_format: TLoaderFileFormat = "puae-jsonl",
```

Comment:
this signature must be generated from the decorated function signature by skipping two first mandatory parameters. I can show you how to do it in mypy ie.
```
@overload
def transformer(
    f: None = ...,
    /,
    data_from: TUnboundDltResource = DltResource.Empty,
    name: TTableHintTemplate[str] = None,
    table_name: TTableHintTemplate[str] = None,
    write_disposition: TTableHintTemplate[TWriteDisposition] = None,
    columns: TTableHintTemplate[TAnySchemaColumns] = None,
    primary_key: TTableHintTemplate[TColumnNames] = None,
    merge_key: TTableHintTemplate[TColumnNames] = None,
    selected: bool = True,
    spec: Type[BaseConfiguration] = None,
    parallelized: bool = False,
    standalone: Literal[True] = True,
) -> Callable[
    [Callable[Concatenate[TDataItem, TResourceFunParams], Any]],
    Callable[TResourceFunParams, DltResource],
]: ...
```
```
 Callable[
    [Callable[Concatenate[TDataItem, TResourceFunParams], Any]],
    Callable[TResourceFunParams, DltResource],
]: ...
```
cuts first parameter from the signature, and you can have more